hpx/runtime/components/binpacking_distribution_policy.hpp

```diff
@@ -0,0 +1,438 @@
+        {}
+
+        /// Create a new \a default_distribution policy representing the given
+        /// set of localities.
+        ///
+        /// \param locs     [in] The list of localities the new instance should
+        ///                 represent
+        /// \param counter_name  [in] The name of the performance counter which
+        ///                      should be used as the distribution criteria
+        ///                      (by default the overall number of existing
+        ///                      instances of the given component type will be
+        ///                      used).
+        ///
+        binpacking_distribution_policy operator()(
+            std::vector<id_type> const& locs,
+            char const* counter_name = default_binpacking_counter_name) const
+        {
+#if defined(HPX_DEBUG)
+            for (id_type const& loc: locs)
+            {
+                HPX_ASSERT(naming::is_locality(loc));
+            }
+#endif
+            return binpacking_distribution_policy(locs, counter_name);
+        }
+
+        /// Create a new \a default_distribution policy representing the given
+        /// locality
+        ///
+        /// \param loc     [in] The locality the new instance should
+        ///                 represent
+        /// \param counter_name  [in] The name of the performance counter which
+        ///                      should be used as the distribution criteria
+        ///                      (by default the overall number of existing
+        ///                      instances of the given component type will be
+        ///                      used).
+        ///
+        binpacking_distribution_policy operator()(id_type const& loc,
+            char const* counter_name = default_binpacking_counter_name) const
+        {
+            HPX_ASSERT(naming::is_locality(loc));
+            return binpacking_distribution_policy(loc, counter_name);
+        }
+
+        /// Create one object on one of the localities associated by
+        /// this policy instance
+        ///
+        /// \param vs  [in] The arguments which will be forwarded to the
+        ///            constructor of the new object.
+        ///
+        /// \returns A future holding the global address which represents
+        ///          the newly created object
+        ///
+        template <typename Component, typename ...Ts>
+        hpx::future<hpx::id_type> create(Ts&&... vs) const
+        {
+            using components::stub_base;
+
+            // handle special cases
+            if (localities_.size() == 0)
+            {
+                return stub_base<Component>::create_async(
+                    hpx::find_here(), std::forward<Ts>(vs)...);
+            }
+            else if (localities_.size() == 1)
+            {
+                return stub_base<Component>::create_async(
+                    localities_.front(), std::forward<Ts>(vs)...);
```

Comment:
Is it intended to only use the first locality here? Shouldn't it rather use the "best"?